arq/jobs.py

```diff
@@ -119,19 +126,29 @@ def __repr__(self):
         return f'<arq job {self.job_id}>'
 
 
-class PickleError(RuntimeError):
+class SerializationError(RuntimeError):
     pass
 
 
-def pickle_job(function_name: str, args: tuple, kwargs: dict, job_try: int, enqueue_time_ms: int):
+def serialize_job(
+    function_name: str,
+    args: tuple,
+    kwargs: dict,
+    job_try: int,
+    enqueue_time_ms: int,
+    *,
+    _serialize: Optional[Callable[[Any], bytes]] = None,
```

Comment:
```suggestion
    serialize: Optional[Callable[[Any], bytes]] = None,
```
since this is explicitly required as a keyword arg, I don't think you need to make it "private".